src/Components/Bodies/CodeBodyUnversionized.tsx

```diff
@@ -165,97 +168,89 @@ class CodeBodyUnversionized extends Component<Props, ICode> {
-                    >{this.extractLabel(this.state.parents[i].code)}</Breadcrumb.Item>)
+            } else {
+                return <div key={i}>
+                    <h5>{translateJson["LBL_" + attribute.toUpperCase()]}</h5>
+                    <p dangerouslySetInnerHTML={{__html: this.state.attributes[attribute]}}/>
+                </div>
             }
-        }
-
-        let translateJson = findJsonService(this.props.params.language);
-        let attributesHtml = [];
-        let i = 1;
+        })
 
-        // Add all non null/empty/undefined code attributes to attributesHtml.
-        for (let attribute in this.state.attributes) {
-            // Skip attributes we do not want to show, like terminal, version, ... or add later (siblings and children).
-            if (skippableAttributes.includes(attribute)) { continue; }
-            // Get value of current attribute.
-            let attributeValue = this.state.attributes[attribute];
-            // Only show attribute if defined, not null or not empty.
-            if (!["", null, undefined].includes(attributeValue)) {
-                if (typeof attributeValue === 'object') {
-                    this.addObjectElement(translateJson, attribute, i, attributeValue, attributesHtml)
-                    i += 1
-                } else {
-                    attributesHtml.push(
-                        <div key={i}>
-                            <h5>{translateJson["LBL_" + attribute.toUpperCase()]}</h5>
-                            <p dangerouslySetInnerHTML={{__html: this.state.attributes[attribute]}}/>
-                        </div>
-                    )
-                    i += 1
-                }
-            }
+        // Add swissmedic number for drugs.
+        if (this.props.params.catalog === "DRUG" && this.props.params.code != 'all') {
+            attributesHtml.push(
+                <div key={attributesHtml.length}>
+                    <h5>{translateJson["LBL_SWISSMEDIC_NR"]}</h5>
+                    <p> {this.state.attributes.auth_number + this.state.attributes.package_code} </p>
+                </div>
+            )
         }
 
         // Add children (subordinate codes).
-        if (this.state.attributes.children) {
-            this.addClickableElement(translateJson, i, 'children', this.state.attributes.children, attributesHtml)
+        let children = this.state.attributes.children;
+        if (children) {
+            attributesHtml.push(this.clickableCodesArray(translateJson, attributesHtml.length, 'children', children))
         }
 
         // Add siblings (similar codes).
-        if(this.state.siblings && !this.state.attributes["children"]) {
-            this.addClickableElement(translateJson, i, "siblings", this.state.siblings, attributesHtml);
+        let siblings = this.state.siblings;
+        if(siblings.length && !children) {
+            attributesHtml.push(this.clickableCodesArray(translateJson, attributesHtml.length, "siblings", siblings))
```

Comment:
Can we get rid of attributesHtml and just render everything directly (using conditional rendering and map..)? It is no good style to compile everything into a variable and then just render this variable: JAX should only be used in one place (return statement) if possible.

Reply:
Also the function e.g. clickableCodesArray might not even be necessary. If they are called multiple times, we should rather create a new component instead of a function that returns a component.

Reply:
The component will look more like the final HTML document and will be more readable like that.